src/main/drivers/transponder_ir_stm32f30x.c

```diff
@@ -104,40 +110,34 @@ void transponderIrHardwareInit(void)
     DMA_InitStructure.DMA_Priority = DMA_Priority_High;
     DMA_InitStructure.DMA_M2M = DMA_M2M_Disable;
 
-    DMA_Init(TRANSPONDER_DMA_CHANNEL, &DMA_InitStructure);
+    DMA_Init(dmaChannel, &DMA_InitStructure);
 
-    TIM_DMACmd(TRANSPONDER_TIMER, TIM_DMA_CC1, ENABLE);
+    TIM_DMACmd(timer, timerDmaSource(timerHardware->channel), ENABLE);
 
-    DMA_ITConfig(TRANSPONDER_DMA_CHANNEL, DMA_IT_TC, ENABLE);
+    DMA_ITConfig(dmaChannel, DMA_IT_TC, ENABLE);
 
 }
 
 void transponderIrDMAEnable(void)
 {
-    DMA_SetCurrDataCounter(TRANSPONDER_DMA_CHANNEL, TRANSPONDER_DMA_BUFFER_SIZE);  // load number of bytes to be transferred
-    TIM_SetCounter(TRANSPONDER_TIMER, 0);
-    TIM_Cmd(TRANSPONDER_TIMER, ENABLE);
-    DMA_Cmd(TRANSPONDER_DMA_CHANNEL, ENABLE);
+    DMA_SetCurrDataCounter(dmaChannel, TRANSPONDER_DMA_BUFFER_SIZE);  // load number of bytes to be transferred
+    TIM_SetCounter(timer, 0);
+    TIM_Cmd(timer, ENABLE);
+    DMA_Cmd(dmaChannel, ENABLE);
 }
 
 void transponderIrDisable(void)
 {
-    GPIO_InitTypeDef GPIO_InitStructure;
-
-    DMA_Cmd(TRANSPONDER_DMA_CHANNEL, DISABLE);
-    TIM_Cmd(TRANSPONDER_TIMER, DISABLE);
-
-    GPIO_StructInit(&GPIO_InitStructure);
-    GPIO_InitStructure.GPIO_Pin = TRANSPONDER_PIN;
-    GPIO_InitStructure.GPIO_Mode = GPIO_Mode_OUT;
-    GPIO_InitStructure.GPIO_OType = GPIO_OType_PP;
-    GPIO_InitStructure.GPIO_PuPd = GPIO_PuPd_DOWN;
-    GPIO_InitStructure.GPIO_Speed = GPIO_Speed_50MHz;
-    GPIO_Init(TRANSPONDER_GPIO, &GPIO_InitStructure);
+    DMA_Cmd(dmaChannel, DISABLE);
+    TIM_Cmd(timer, DISABLE);
+
+    IOInit(transponderIO, OWNER_TRANSPONDER, 0);
+    IOConfigGPIOAF(transponderIO, IO_CONFIG(GPIO_Mode_AF, GPIO_Speed_50MHz, GPIO_OType_PP, GPIO_PuPd_DOWN), timerHardware->alternateFunction);
+
 #ifdef TRANSPONDER_INVERTED
-    digitalHi(TRANSPONDER_GPIO, TRANSPONDER_PIN);
+    IOHi(transponderIO);
 #else
-    digitalLo(TRANSPONDER_GPIO, TRANSPONDER_PIN);
+    IOLo(transponderIO);
```

Comment:
Should probably use the inversion based on the timer array - otherwise the inversion on init and this inversion could be out of step.